python/pycrdt/_awareness.py

```diff
@@ -0,0 +1,182 @@
+                elif client_meta is not None and state is None:
+                    removed.append(client_id)
+                elif state is not None:
+                    if state != prev_state:
+                        filtered_updated.append(client_id)
+                    updated.append(client_id)
+
+        changes = {
+            "added": added,
+            "updated": updated,
+            "filtered_updated": filtered_updated,
+            "removed": removed,
+            "states": states,
+        }
+
+        # Do not trigger the callbacks if it is only a keep alive update
+        if added or filtered_updated or removed:
+            for callback in self._subscriptions:
+                callback(changes)
+
+        return changes
+
+    def get_local_state(self) -> dict[str, Any]:
+        return self._states.get(self.client_id, {})
+
+    def set_local_state(self, state: dict[str, Any]) -> None:
+        """
+        Update the local state and meta.
+        This function call the on_change() callback (if provided), with the states
+        formatted (bytes) as argument.
+
+        Args:
+            state: The dictionary representing the state.
+        """
+        timestamp = int(time.time() * 1000)
+        clock = self.meta.get(self.client_id, {}).get("clock", -1) + 1
+        self._states[self.client_id] = state
+        self.meta[self.client_id] = {"clock": clock, "last_updated": timestamp}
+        # Build the message to broadcast, with the following information:
+        # - message type
+        # - length in bytes of the updates
+        # - number of updates
+        # - for each update
+        #   - client_id
+        #   - clock
+        #   - length in bytes of the update
+        #   - encoded update
+        msg = json.dumps(state, separators=(",", ":")).encode("utf-8")
+        msg = write_var_uint(len(msg)) + msg
+        msg = write_var_uint(clock) + msg
+        msg = write_var_uint(self.client_id) + msg
+        msg = write_var_uint(1) + msg
+        msg = write_var_uint(len(msg)) + msg
+        msg = write_var_uint(YMessageType.AWARENESS) + msg
+
+        if self.on_change:
+            self.on_change(msg)
+
+    def set_local_state_field(self, field: str, value: Any) -> None:
+        """
+        Set a local state field.
+
+        Args:
+            field: The field to set (str)
+            value: the value of the field
+        """
+        current_state = self.get_local_state()
+        current_state[field] = value
+        self.set_local_state(current_state)
+
+    def observe(self, callback: Callable[[dict[str, Any]], None]) -> None:
+        """
+        Subscribes to awareness changes.
+
+        Args:
+            callback: Callback that will be called when the document changes.
+        """
+        self._subscriptions.append(callback)
+
+    def unobserve(self) -> None:
+        """
+        Unsubscribes to awareness changes.
+
+        This method removes all the callbacks.
```

Comment:
```suggestion
        Unsubscribes to awareness changes. This method removes all the callbacks.
```
BTW maybe there should be a way to unregister a single callback, instead of all of them?